Add user mode CLI command

Dexter2038 opened the pull request:
I needed it, so I added it.

Tested, everything is working.

![image](https://github.com/user-attachments/assets/b53aafba-5fb3-47fd-8b2f-50525f53af89)

Dexter2038 committed:
Add user mode CLI command

Dexter2038 committed:
Fix incompatible functions

guillaumeboehm commented:
Hey! Thanks for the PR, sorry it took me so long to come back to it, do you mind rebasing on the latest main? If you don't use this anymore tell me so I'll take care of it myself ;)

Dexter2038 commented:
Thanks for getting back to me! I’m happy to rebase my branch since I’m still using my modified version of the app.

Dexter2038 committed:
Replace strcmp with string operators

Dexter2038 commented:
<img width="1193" height="1020" alt="250815_14h21m48s_screenshot" src="https://github.com/user-attachments/assets/6d275ef1-de17-4c1b-83a7-23e41984845e" />
Changed some commits' names and authorship, fixed conflicts and fixed some errors. Pinned screenshot of successfully running program. It builds, it works. At least on my machine. Waiting to finally merge

guillaumeboehm commented on src/cli.cpp:
```diff
@@ -58,3 +58,65 @@ void CLI::setCoolerBoost(Options::State state){
+    }
+    fprintf(stdout, "Changing user mode to %s mode\n", ( text_mode.c_str() ));
+    operate.setUserMode(user_mode);
+    operate.updateEcData();
```
Prefer this here
```
if(!operate.updateEcData()) {
    fprintf(stderr, "Failed to update EC data for Cooler Boost\n");
}
```

guillaumeboehm reviewed (changes requested):
Alright, just a tiny improvement and we're good

Dexter2038 commented:
Oh, I accidentally closed it..

Dexter2038 committed:
Print error on fail of upate ec data

Dexter2038 commented:
Added print on failed EC data update

Dexter2038 commented:
Should I change "user mode" to "User Mode" for consistency? The code prints "Cooler Boost" and not "cooler boost", so I guess it would be better if naming would be the same, right?

Dexter2038 committed:
Rename user mode to User Mode

Dexter2038 commented:
I did change it.

guillaumeboehm commented:
> Should I change "user mode" to "User Mode" for consistency? The code prints "Cooler Boost" and not "cooler boost", so I guess it would be better if naming would be the same, right?

Eh it was fine but ok ^^
